tutorials/tts/FastPitch_MultiSpeaker_Pretraining.ipynb

```diff
@@ -0,0 +1,867 @@
+    "+init_from_pretrained_model=\"tts_en_fastpitch\" \\\n",
+    "sample_rate={sample_rate} \\\n",
+    "train_dataset={train_manifest} \\\n",
+    "validation_datasets={valid_manifest} \\\n",
+    "sup_data_types=\"['align_prior_matrix', 'pitch', 'speaker_id','reference_audio']\" \\\n",
+    "sup_data_path={suppdir} \\\n",
+    "pitch_mean={pitch_mean} \\\n",
+    "pitch_std={pitch_std} \\\n",
+    "phoneme_dict_path={normdir}/cmudict-0.7b_nv22.10 \\\n",
+    "heteronyms_path={normdir}/heteronyms-052722 \\\n",
+    "model.speaker_encoder.lookup_module._target_=\"nemo.collections.tts.modules.submodules.SpeakerLookupTable\" \\\n",
+    "model.speaker_encoder.lookup_module.n_speakers=5 \\\n",
+    "model.speaker_encoder.gst_module._target_=\"nemo.collections.tts.modules.submodules.GlobalStyleToken\" \\\n",
+    "model.input_fft.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.output_fft.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.duration_predictor.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.pitch_predictor.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.alignment_module.condition_types=\"['add']\" \\\n",
+    "model.train_ds.dataloader_params.batch_size=8 \\\n",
+    "model.validation_ds.dataloader_params.batch_size=8 \\\n",
+    "model.train_ds.dataloader_params.num_workers=8 \\\n",
+    "model.validation_ds.dataloader_params.num_workers=8 \\\n",
+    "model.train_ds.dataset.max_duration=20 \\\n",
+    "model.validation_ds.dataset.max_duration=20 \\\n",
+    "model.validation_ds.dataset.min_duration=0.1 \\\n",
+    "+model.text_tokenizer.add_blank_at=True \\\n",
+    "exp_manager.exp_dir={logsdir} \\\n",
+    "+exp_manager.create_wandb_logger=True \\\n",
+    "+exp_manager.wandb_logger_kwargs.name=\"tutorial-FastPitch-pretrain-multispeaker\" \\\n",
+    "+exp_manager.wandb_logger_kwargs.project=\"NeMo\" \\\n",
+    "trainer.max_epochs=20 \\\n",
+    "trainer.check_val_every_n_epoch=20 \\\n",
+    "trainer.log_every_n_steps=1 \\\n",
+    "trainer.devices=-1 \\\n",
+    "trainer.strategy=ddp \\\n",
+    "trainer.precision=32 \\\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "73bc625f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# e.g. NeMoTTS_logs/FastPitch/Y-M-D_H-M-S/checkpoints/FastPitch--val_loss=XXX-epoch=XXX-last.ckpt\n",
+    "last_checkpoint_dir = sorted(list([i for i in (Path(logsdir) / \"FastPitch\").iterdir() if i.is_dir()]))[-1] / \"checkpoints\"\n",
+    "YOUR_PRETRAINED_FASTPITCH_CHECKPOINT = list(last_checkpoint_dir.glob('*-last.ckpt'))[0]\n",
+    "YOUR_PRETRAINED_FASTPITCH_CHECKPOINT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "711a9913",
+   "metadata": {},
+   "source": [
+    "# 2. Fine-tune HiFiGAN on multi-speaker data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88364f18",
+   "metadata": {},
+   "source": [
+    "## a. Dataset Preparation"
```

Comment:
Can you use https://github.com/NVIDIA/NeMo/blob/main/scripts/dataset_processing/tts/generate_mels.py or https://github.com/NVIDIA/NeMo/blob/main/scripts/dataset_processing/tts/resynthesize_dataset.py ?

Reply:
Do I need to modify `generate_mels.py` and `resynthesize_dataset.py` to support reference audio? The inference function needs the reference audio, so I don't use this two scripts currently.

Reply:
Yes. If I remember correctly the two scripts are similar (`generate_mels.py` uses predicted pitch/energy while `resynthesize_dataset.py` uses ground truth pitch/energy). Supporting reference audio would make more sense in `resynthesize_dataset.py`, but depends on how exactly you are doing the fine-tuning.

Reply:
For `FastPitch_Adapter_Finetuning.ipynb`, if we use `resynthesize_dataset.py`, we can't load adapter weights in `SpectrogramGenerator`; if we use `generate_mels.py`, we need to add `load_adapters`. Additionally, because of the new speaker encoder, we also need input `reference audio` and make input `speaker` optional. Do you think we can generate mel-spectrograms without using the script in the notebook?

Reply:
If there are significant problems with reusing the previous scripts, then we can leave it in the notebook.

Figuring out how to make inference logic consistent across models is a different project (eg. https://github.com/NVIDIA/NeMo/pull/6096) and I don't think extending base.SpectrogramGenerator or adding a 3rd duplicate inference script is a good idea at this point.